### PR TITLE
Set layout height to 100dvh with fallback to 100vh

### LINKED ISSUE #2110: On mobile, homepage content switcher gets cut-off at the bottom
## Description

The bottom part of the homepage is cut off on mobile. On an Android phone, it is worse on Chrome than in Firefox:

## Reproduction

1. Open http://staging.openverse.org on your mobile phone
2. Try changing the search type by clicking on the content switcher button inside the search bar.

### Firefox
On Firefox, if you try scrolling on the homepage, the footer is visible:
Chrome homepage screenshot
But the last item in the content switcher is still cut off:
Chrome homepage screenshot
### Chrome

WordPress logo in the bottom of the footer is cut off:
Chrome homepage screenshot
The last item in content switcher is cut off:
Chrome homepage screenshot

## Environment

- Device: Samsung phone
- OS: Android
- Browser: Chrome, Firefox

## Additional context
The issue is caused by the way mobile browsers handle the height in `vh` units and hiding /showing the address bar.

One quick fix would be to add padding at the bottom of the homepage footer and the content switcher with a height equal to the height of the address bar.

Another would be to set the height to "100%". However, due to the way Nuxt layout is nested, we would need to add the height value to several nested divs. Also, 100% height is the height of the viewport without the height of the address bar. This means that when the address bar is hidden, there will probably be some space at the bottom.

@WordPress/openverse-frontend, what do you think is the best solution here?

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2110 by @obulat

## Description

This PR fixes the bug where the bottom of the screen is cut-off and hidden behind the address bar on mobiles by using `dvh` units for screen height.
Currently, the support for them is below 80%, so as a fallback, we set the height to 100 vh but also add a bottom padding to the modals to make sure that clickable elements are not hidden.

## Testing Instructions

Run the app and test it on a mobile. You should see the WordPress logo in the footer. When you input something in the hompage searchbar, even the bottom part of the screen should remain yellow. When you open modal (content switcher on homepage), none of it should be hidden.

I don't have a phone to test the fallback, it would be nice if someone could test it on a real device.
## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
This does not fix the issue of the background getting cut off.

File: src/layouts/default.vue [hunk 1]
```diff
@@ -207,9 +207,12 @@ export default {
 </script>
 
 <style scoped>
+.app {
+  @apply min-h-screen min-h-[100dvh];
+}
 .sidebar {
   /* Header height above md is 80px plus 1px for bottom border */
-  height: calc(100vh - 81px);
+  @apply h-[calc(100vh-81px)] h-[calc(100dvh-81px)];
```
Comment (zackkrida) [inline src/layouts/default.vue:215]:
```suggestion
  @apply h-[calc(100vh - 81px)] h-[calc(100dvh - 81px)];
```

These calc statements need spaces between the operators to work.

**Comment by zackkrida** [comment]
@obulat not related to the changes in this PR, I noticed that desktop safari needs the homepage to explicitly have `height: 100vh` and not `min-height` to display properly, otherwise it looks like this:

> initial 


CleanShot 2023-02-01 at 07 41 09

> after scrolling

CleanShot 2023-02-01 at 07 41 12


I think it's some kind of bug related to min-height, flex containers, and sticky flex children 😢 

I will test this on a browser that doesn't support `dvh` and give a final review later today!

File: src/layouts/default.vue (hunk 1 above)
Comment (obulat) [reply src/layouts/default.vue:215]: Tailwind says they need underscores: https://tailwindcss.com/docs/adding-custom-styles#handling-whitespace
But the arbitrary values can also use no whitespace at all: https://stackoverflow.com/questions/65976223/how-to-use-calc-in-tailwind-css. I think the "no whitespace" version looks cleaner than the "undescores" one.

File: src/components/VFourOhFour.vue [hunk 2]
```diff
@@ -2,7 +2,9 @@
   <div
     class="error relative flex flex-col overflow-x-hidden"
     :class="
-      isNewHeaderEnabled ? 'flex-grow px-6 sm:px-0' : 'min-h-screen bg-yellow'
+      isNewHeaderEnabled
+        ? 'flex-grow px-6 sm:px-0'
```
Comment (zackkrida) [inline src/components/VFourOhFour.vue:6]: 👍 makes sense

File: src/components/VModal/VModalContent.vue [hunk 3]
```diff
@@ -244,4 +247,15 @@ export default defineComponent({
 .modal-content-light {
   @apply bg-white text-dark-charcoal;
 }
+/*
+For mobiles that do not support dvh units, we add a fallback padding
+to the modal content to make sure that no clickable elements is hidden
+by the address bar.
```
Comment (zackkrida) [inline src/components/VModal/VModalContent.vue:253]:
```suggestion
to the modal content to make sure that no clickable elements are hidden
by the address bar.
```

File: src/layouts/default.vue (hunk 1 above)
Comment (zackkrida) [reply src/layouts/default.vue:215]: Oh I see, tailwind will produce the correctly-spaced result in the css. I agree this looks cleaner.

**Review by zackkrida — APPROVED** [review]
I think this LGTM! I'm seeing things look right on desktop and mobile in chrome, firefox, and safari.

**Review by dhruvkb — APPROVED** [review]
This looks great now.

**Pull request merged**